### ISSUE TITLE
Make quickFilters compatible with Thunderbird 91

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/quickFilters#48]
Thunderbird 86 introduced multi-process browsing (e10s a.k.a. electrolysis) which creates some compatibility issues with some functions.

Quick patch version for 86.0b3:
[quickFilters-wx-5.2pre6.zip](https://github.com/RealRaven2000/quickFilters/files/6009522/quickFilters-wx-5.2pre6.zip)

To install, download the xip file and drag into Thunderbird Add-ons Manager.

Ongoing bugfixes for beta versions will go here.

**Comment by RealRaven2000** [comment]
Importing filters and filter trouble shooter wasn't working. Also any filters that have no search condition at all will disrupt the troubleshooting process. 

Patched version:
[quickFilters-wx-5.2pre20.zip](https://github.com/RealRaven2000/quickFilters/files/6015193/quickFilters-wx-5.2pre20.zip)

**Comment by aprojanos71** [comment]
Super!
It works! I see the names of the faulty filters and the backup also works fine. 
Thx!

**Comment by aprojanos71** [comment]
Merge does not work ...

**Comment by RealRaven2000** [comment]
> Merge does not work ...

thanks for finally using github for bug reporting. What exactly doesn't work - how do you invoke it?

When you merge filters they have to have the same (main) action - e.g. move mails to the same folder.
You can select 2 filters with the same action and do merge manually or you are prompted to merge when you add new condition to an existing filter - e,g, dropping an email to a folder that already has filters addressing it.

**Comment by RealRaven2000** [comment]
I tested in Tb86.0b3 with the assistant (assistant is active, I already had a filter that moves mail to pinterest) I moved a mail to the folder "pinterest" and it came up with the merge suggestion for the existing filter. I was able to successfully merge in the condition "recommendations@explore.pinterest.com" which was not pre-existing in the filter. (the condition with @pinterest already existed)

![image](https://user-images.githubusercontent.com/6314674/108601270-b771b000-7393-11eb-8d17-fcf5c50dc9ac.png)


![image](https://user-images.githubusercontent.com/6314674/108601232-898c6b80-7393-11eb-9b51-0a825561939f.png)

**Comment by aprojanos71** [comment]
When I select two filters and want to merge, it does nothing.
The two filters do the same thing - [move to spam] - but do nothing. 

Uncaught TypeError: stCollection.Count is not a function                                                                          qFilters-utils.js:1236:63
    copyTerms chrome://quickfilters/content/qFilters-utils.js:1236
    merge chrome://quickfilters/content/qFilters-list.js:423
    oncommand chrome://messenger/content/FilterListDialog.xhtml:1
[Learn More]
    copyTerms chrome://quickfilters/content/qFilters-utils.js:1236
    merge chrome://quickfilters/content/qFilters-list.js:423
    oncommand chrome://messenger/content/FilterListDialog.xhtml:1

**Comment by aprojanos71** [comment]
![ScreenShot098](https://user-images.githubusercontent.com/78041865/108601648-e4c35b80-739d-11eb-9ca1-d97a13649c34.jpg)

**Comment by RealRaven2000** [comment]
Try this version:

[quickFilters-wx-5.2pre24.zip](https://github.com/RealRaven2000/quickFilters/files/6015502/quickFilters-wx-5.2pre24.zip)

There was another instance of `searchTerms.Count()` which is not supported in Thunderbird 86, I think they changed the type of the searchTerms attribute from `nsIMutableArray` to an ` Array<nsIMsgSearchTerm>`. See code:
[comm-esr78](https://searchfox.org/comm-esr78/source/mailnews/extensions/mailviews/nsIMsgMailView.idl#18)
[comm-central](https://searchfox.org/comm-central/source/mailnews/extensions/mailviews/nsIMsgMailView.idl#17)

Still a side effect from not using APIs I think changes like these will be even more frequent this year:

https://bugzilla.mozilla.org/show_bug.cgi?id=857230

**Comment by aprojanos71** [comment]
Yes, it works!
Thx!

**Comment by RealRaven2000** [comment]
Latest version, including the "Sort search terms" [feature](https://github.com/RealRaven2000/quickFilters/issues/11) on the edit filter dialog:

[quickFilters-wx-5.2pre72.zip](https://github.com/RealRaven2000/quickFilters/files/6240040/quickFilters-wx-5.2pre72.zip)

Install by downloading the zip file and then dropping it into Thunderbird Add-ons Manager. Compatible from Thunderbird 78 forward.

**Comment by RealRaven2000** [comment]
Problem on the support tab in Thunderbird 91.0b1 with descriptions for the buttons "sliding down" to the bottom:
![image](https://user-images.githubusercontent.com/6314674/126064815-a6ee464a-bb44-4a1a-b465-9aa5d10016b8.png)

Also, when trying to open the debug settings via right-click, the about:config panel is not opened anymore:
![image](https://user-images.githubusercontent.com/6314674/126065022-c4c94d0a-876d-40ae-881e-cc334cf297f4.png)

**Comment by RealRaven2000** [comment]
Latest version see below
Changes:
- removed legacy localization (bundle string) and removed 108 translation (dtd + properties) files after consolidating into messages.json (1 file per language)
- moved licensing mechanism into crypto module (experimental API)
- added colorized icons on filter list toolbar [issue #67]
- use background script for all communication between windows (via notification API)
- Remember Copied filters List to insert to other Accounts multiple times [issue #20] 

[quickFilters-wx-5.3pre136.zip](https://github.com/RealRaven2000/quickFilters/files/6884595/quickFilters-wx-5.3pre136.zip)

To install, download the zip file and then drag that file into Thunderbird Add-ons Manager.

**Comment by RealRaven2000** [comment]
Latest version see below
Changes:

- fixed layout on the Support screen and added button to Display Splash Screen
- removed use of fixIterator (deprecated function in Tb91) from `findInboxFromRoot()` function.
- removed legacy code for displaying version number in options / update check
- slight layout changes in update splash screen
- removed links to mozdev.org for legacy bugs

[quickFilters-wx-5.3pre154.zip](https://github.com/RealRaven2000/quickFilters/files/6902920/quickFilters-wx-5.3pre154.zip)

To install, download the zip file and then drag that file into Thunderbird Add-ons Manager.

**Comment by RealRaven2000** [comment]
Latest Version
Changes:

- Restricted  special search types in filter list to licensed users
- Fixed style of [Buy License] button at bottom of settings dialog
- Fixed icons in sliding notifications
- Fixed missing "about:config" dialogs

[quickFilters-wx-5.3pre167.zip](https://github.com/RealRaven2000/quickFilters/files/6924424/quickFilters-wx-5.3pre167.zip)

To install, download the zip file and then drag that file into Thunderbird Add-ons Manager.

**Comment by RealRaven2000** [comment]
I uploaded a new release to ATN this morning, see https://github.com/RealRaven2000/quickFilters/releases/tag/5.3.1
In this one, I do not set strict_max_version (mainly because there are no "legal" values greater than 91.0a1 available to the current validator) - I only have tested it as far as the current 92 daily. But it should be fully compatible with 91.0 ESR as far as I can tell. Leaving this issue open for Tb91 related regressions until the majority of users are migrated to it.

I also finally created the product for a [quickFilters domain license on the fastspring shop](http://sites.fastspring.com/quickfolders/product/quickfiltersdomainlicense). I just need to add the decryption keys for domain licenses in the next quickFilters release
